app/controllers/documents_controller.rb

```diff
@@ -0,0 +1,66 @@
+class DocumentsController < ApplicationController
+    before_action :authenticate_user!, except: [:public_show]
+
+  def index
+    @documents = current_user.documents
+    @documents_count = @documents.count
+  end
+
+  def new
+    @document = current_user.documents.new
+  end
+
+  def create
+    @document = current_user.documents.new(document_params)
+    if @document.save
+      redirect_to documents_path, notice: 'File uploaded successfully.'
+    else
+      # Handle case where the form submission is invalid
+      flash.now[:alert] = 'Error: Unable to save document.'
+      render :new
+    end
+  end
+
+  def download
+    @document = Document.find(params[:id])
+    redirect_to rails_blob_path(@document.file, disposition: 'attachment')
+  end
+
+  def destroy
+    @document = Document.find(params[:id])
+    # Delete the attached file
+    @document.file.purge if @document.file.attached?
```

Comment:
what happens in the case where file is purged
but db record deletion failed due to any reason

Reply:
modified code to purge after save success